### DISCUSSION TITLE
Expect a function not to be called?

### DISCUSSION THREAD

**Discussion by dreambucket13** [ThrowTheSwitch/Ceedling#701] [Ideas]
So in this scenario:

```
if (condition)
{
void myFunc();
}
```

How do I check that myFunc() _is not_ called if condition evaluates to false?

**Comment by mvandervoord** [comment]
Good news. You don't have to do anything except make sure that `condition` evaluates to `false`. If `myFunc` *does* get called, then CMock will notice that it was called when it wasn't expected and will report the failure as `myFunc called more times than expected`. (because it was called at least once, and we expected 0 calls)

**Reply by Letme** [reply]
Reading that, I just remembered, that you still need to include the mock header file and use setup/verify calls in setup/teardown of your test file for this to be reported.

**Reply by dreambucket13** [reply]
Are there any reference docs out there on how to do that? Cmock_Summary doesn't seem to cover that.

**Reply by mvandervoord** [reply]
I assume since this is on the Ceedling forum and not the CMock forum that you're using Ceedling. The setup/verify calls are handled for you by ceedling and you don't need to worry about them. You just have to include the mock.

To get it to link at all, you'll need to include either the mock or the real thing. 

I can definitely add this to the CMock summary. :) Leave this issue open and it'll happen!

**Reply by dreambucket13** [reply]
Ah yes I am indeed using Ceedling. Thanks!

**Comment by dreambucket13** [comment]
Update - I am clearly doing something wrong. I made a very simple project and this works as I expected. 

https://github.com/dreambucket13/ceedling-playground

**Reply by Letme** [reply]
You probably included real header file before the mock
